src/main/java/seedu/clinic/logic/parser/MacroParser.java

```diff
@@ -0,0 +1,30 @@
+package seedu.clinic.logic.parser;
+
+import java.util.Arrays;
+import java.util.Optional;
+
+import seedu.clinic.model.Model;
+import seedu.clinic.model.macro.Macro;
+
+public class MacroParser {
+
+    /**
+     * Converts user input into the corresponding command string if a macro is identified to be used.
+     *
+     * @param model the model which contains the user macros
+     * @param userInput full user input string
+     * @return the command string saved by the macro, or the original input string no macro is used
```

Comment:
I think there is missing a word here, should be "original input string when no macro is used"

Reply:
Good spot! Thanks.